src/animation/mod.rs

```diff
@@ -58,45 +60,57 @@ impl PlayingAnimation2D {
     ///
     /// Note: An animation with `RepeatAnimation::Forever` will never finish.
     #[inline]
-    pub fn is_finished(&self) -> bool {
+    pub fn finished(&self) -> bool {
         match self.repeat {
             RepeatAnimation::Forever => false,
             RepeatAnimation::Never => self.completions >= 1,
             RepeatAnimation::Count(n) => self.completions >= n,
         }
     }
 
+    /// Check if the animation has just finished.
+    ///
+    /// Note: This needs to be called in the `bevy_app::main_schedule::Update` schedule.
+    #[inline]
+    pub fn just_finished(&self) -> bool {
+        self.completions_this_update > 0
+    }
```

Comment:
Most importantly this does not care about the animation repetition behaviour. I think that information is important (e.g. to not change to another animation, if the source animation is Forever), but maybe it should be another interface then. 
Open for suggestions.

Reply:
It's certainly interesting, there have been a few instances were I wanted to get the finished state of a looping animation (playing footsteps on walk animations). I would just go roll with this and see how it works in action, usually you will make sure that a certain state is active anyways and I think this is fine, though there can always be some ugly edge cases.

If nothing else we could do something like

```
pub fn just_finished(self) -> bool {
    self.completions_this_update > 0 && self.repeat != RepeatAnimation::Forever
}

pub fn just_finished_raw(self) -> bool {
    self.completions_this_update > 0
}
```

though the user can also do this if he really wanted/needed to.

Reply:
I ended up going with finished and just_finished for existing behaviour based on repitition behaviour. And I added a new just_finished_cycle for checking a single completion.

Reply:
I really like this implementation.